src/evaluating_students/router.js

```diff
@@ -1,14 +1,354 @@
+            res.sendStatus(400);
+            return
+        }
+
+        teacher.firstName = req.body.firstName;
+        teacher.lastName = req.body.lastName;
+        teacher.save()
+            .then((savedTeacher) => {
+                res.status(201);
+                res.json(savedTeacher)
+            })
+            .catch((err) => res.sendStatus(500));
+    })
+
+    .delete((req, res) => TeacherModel.remove({})
+        .then((deleted) => {
+            res.status(200);
+            res.json(deleted)
+        })
+        .catch(() => res.sendStatus(500))
+    );
+
+
+/**
+ * @swagger
+ *  paths:
+ *      /students:
+ *         get:
+ *           summary: Return list of all students
+ *           responses:
+ *              '200':
+ *                  description: "List of all students"
+ *              '500':
+ *                  description: "Database query failed"
+ *         put:
+ *           summary: Not implemented.
+ *           responses:
+ *              '501':
+ *                  description: "Not implemented"
+ *         post:
+ *           summary: Add student to student collection
+ *           parameters:
+ *              -   in: body
+ *                  name: student
+ *                  description: The student to create.
+ *                  schema:
+ *                      type: object
+ *                      required:
+ *                          -   firstName
+ *                          -   lastName
+ *                      properties:
+ *                          firstName:
+ *                              type: string
+ *                          lastName:
+ *                              type: string
+ *           responses:
+ *              '201':
+ *                  description: "[Saved document student]"
+ *              '400':
+ *                  description: "Bad request"
+ *              '500':
+ *                  description: "Failed to save document"
+ *         delete:
+ *           summary: Deletes all documents in the student collection
+ *           responses:
+ *              '200':
+ *                  description: "[Count of deleted documents]"
+ *              '500':
+ *                  description: "Failed to delete document"
+ */
+router.route('/students')
+    .get((req, res) => {
+        StudentModel.find({})
+            .then((allStudents) => res.json(allStudents))
+            .catch(() => res.sendStatus(500))
+    })
+
+    .put((req, res) => res.sendStatus(501))
+
+    .post((req, res) => {
+        let student = new StudentModel();
+
+        const firstName = req.body.firstName;
+        const lastName = req.body.lastName;
+
+        if (firstName === undefined || lastName === undefined) {
+            res.sendStatus(400);
+            return
+        }
+
+        student.firstName = req.body.firstName;
+        student.lastName = req.body.lastName;
+        student.save()
+            .then((savedStudent) => {
+                res.status(201);
```

Comment:
Aan elkaar

Reply:
Aan elkaar?

Reply:
res.status().json();

Moet dit trouwens geen 303 redirect naar de GET request zijn i.v.m. cqrs?

Reply:
Aha.

En nee? Er is nog nergens cqrs toegepast, laat staan dat dit er iets mee te maken heeft. Het is een post request om een student aan te maken?

Reply:
Ah ok, ik dacht dat volgens de opdracht get requests alleen data zouden teruggeven en post/delete/put redirects naar get requests enzo 🤔